namerd/iface/control-http/src/main/scala/io/buoyant/namerd/iface/HttpControlService.scala

```diff
@@ -120,91 +120,100 @@ class HttpControlService(storage: DtabStore) extends Service[Request, Response]
-
-        case Method.Put =>
-          req.contentType.flatMap(DtabCodec.byContentType) match {
-            case Some(codec) =>
-              codec.read(req.content) match {
-                case Return(dtab) =>
-                  req.headerMap.get("If-Match") match {
-                    case Some(stamp) =>
-                      val buf = Buf.ByteArray.Owned(Base64StringEncoder.decode(stamp))
-                      storage.update(ns, dtab, buf).transform {
-                        case Return(_) =>
-                          Future.value(Response(Status.NoContent))
-                        case Throw(e: DtabNamespaceDoesNotExist) =>
-                          Future.value(Response(Status.NotFound))
-                        case Throw(e: DtabVersionMismatchException) =>
-                          Future.value(Response(Status.PreconditionFailed))
-                        case Throw(_) =>
-                          Future.value(Response(Status.InternalServerError))
-                      }
-                    case None =>
-                      val rsp = Response(Status.PreconditionRequired)
-                      rsp.contentString = "If-Match must be used."
-                      rsp.contentType = MediaType.PlainText
-                      Future.value(rsp)
-                  }
-
-                // invalid dtab
-                case _ => Future.value(Response(Status.BadRequest))
-              }
-
-            // invalid content type
-            case _ => Future.value(Response(Status.BadRequest))
-          }
-
-        case Method.Post =>
-          req.contentType.flatMap(DtabCodec.byContentType(_)) match {
-            case Some(codec) =>
-              codec.read(req.content) match {
-                case Return(dtab) =>
-                  storage.create(ns, dtab).transform {
-                    case Return(_) => Future.value(Response(Status.NoContent))
-                    case _ => Future.value(Response(Status.InternalServerError))
-                  }
-
-                // invalid dtab
-                case _ => Future.value(Response(Status.BadRequest))
-              }
-
-            // invalid content type
-            case _ => Future.value(Response(Status.BadRequest))
-          }
-
-        // invalid method
-        case _ => Future.value(Response(Status.NotFound))
-      }
+  private[this] def handleList(): Future[Response] =
+    storage.list().map { namespaces =>
+      val rsp = Response()
+      rsp.contentType = MediaType.Json
+      rsp.content = Json.write(namespaces)
+      rsp
+    }
 
-    // invalid uri
-    case _ => Future.value(Response(Status.NotFound))
-  }
+  private[this] def handleGetDtab(ns: String, req: Request): Future[Response] =
+    getDtab(ns).map {
+      case Some(dtab) =>
+        val rsp = Response()
+        val (contentType, codec) = DtabCodec.accept(req.accept).getOrElse(DtabCodec.default)
+        rsp.contentType = contentType
+        rsp.headerMap.add("ETag", versionString(dtab.version))
```

Comment:
minor: finagle-http appears to have "ETag" as a constant in `Fields`; might as well use that